.github/workflows/dogfood.yml

```diff
@@ -0,0 +1,52 @@
+name: dogfood merge
+# This workflow is triggered on pushes to master and dogfood branches
+on:
+  push:
+    branches:
+      - master
+      - 'dogfood/*'
+
+env:
+  SLACK_CHANNEL: team-lang-dotnet
+  SLACK_USERNAME: Dogfood build action
+
+jobs:
+  dogfood_merge:
+    runs-on: ubuntu-latest
+    name: Update dogfood branch
+    permissions:
+      id-token: write # required for SonarSource/vault-action-wrapper
+      contents: write # required to grant GITHUB_TOKEN writing permission
```

Comment:
Educational: why do we need the `contents` permission?
sonar-java doesn't seem to require it, and has a very similar yml, see [here](https://github.com/SonarSource/sonar-java/blob/master/.github/workflows/dogfood.yml).

Reply:
I have no idea. The python one uses it. I'll remove it and we see if it works.